aptos-move/move-examples/dao/nft_dao/sources/nft_dao.move

```diff
@@ -0,0 +1,639 @@
+    }
+
+    /// The collection should only contains NFTs, where all token name only has 1 maximal and immutable
+    /// The total supply is fixed with the token names.
+    struct GovernanceToken has store {
+        /// The creator address of this NFT collection
+        creator: address,
+        /// The collection name
+        collection: String,
+        /// Eligible token_ids that can vote. Value of the table is the weights of the token
+        member_tokens: Table<TokenId, u64>
+    }
+
+    /// All proposals
+    struct Proposals has key {
+        proposals: Table<u64, Proposal>,
+    }
+
+    /// Store the general information about a proposal
+    struct Proposal has copy, drop, store {
+        /// Name of the proposal, limiting to 64 chars
+        name: String,
+        /// Description of the proposal, limiting to 512 chars
+        description: String,
+        /// The name of function to be executed
+        function_name: String,
+        /// The function arguments to be exectued
+        function_args: PropertyMap,
+        /// The start time of the voting
+        start_time_sec: u64,
+        /// Proposal results, unresolved, passed, not passed
+        resolution: u8,
+    }
+
+    struct ProposalVotingStatistics has key {
+        proposals: Table<u64, VotingStatistics>,
+    }
+
+    struct VotingStatistics has store {
+        /// Total yes votes
+        total_yes: u64,
+        /// Total no notes
+        total_no: u64,
+        /// Token voted yes
+        yes_votes: BucketTable<TokenId, address>, // address is the original voter's address for keeping a record of who voted
+        /// Token voted no
+        no_votes: BucketTable<TokenId, address>,
+    }
+
+    struct RegisteredAccounts has key {
+        /// Key is the address of DAO creator's account and value is the signer capability of its resource account
+        /// Only one DAO per admin address for now.
+        accounts_signer_caps: Table<address, SignerCapability>,
+        res_acct_to_src_acct: Table<address, address>,
+    }
+
+    /// Initialize the DAO platform
+    fun init_module(account :&signer) {
+        move_to(
+            account,
+            RegisteredAccounts {
+                accounts_signer_caps: table::new(),
+                res_acct_to_src_acct: table::new(),
+            }
+        );
+    }
+
+    /// Creator creates a DAO on the platform
+    public entry fun create_dao(
+        admin: &signer,
+        name: String,
+        nft_creator_address: address, // this is the controlled nft account
+        threshold_numerator: u64,
+        threshold_denominator: u64,
+        voting_duration: u64,
+        creator: address, // this is the creator address of goverance token
+        collection_name: String,
+        min_proposal_weight: u64,
+    ) acquires RegisteredAccounts {
+
+        // create a resource account
+        let (res_signer, res_cap) = account::create_resource_account(admin, bcs::to_bytes(&name));
```

Comment:
I'd suggest adding additional seeds for domain separation to avoid potential conflicts with other resource accounts that might be created.

For example: https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-framework/sources/staking_contract.move#L601

Reply:
updated